Pick and place planning

avalenzu opened the pull request:
This PR re-factors the pick-and-place demo to use IK traj instead of `ConstraintRelaxingIk`.

<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/7567)
<!-- Reviewable:end -->

avalenzu commented:
+(status: curate commits before merging)  +@naveenoid for feature review.

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L-mp0xgrBwt90v4oqOy:bz1e5wi)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
-@naveenoid +@sammy-tri for feature review 

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L-n_1lv_3T7XM-W8xe-:b86il4y)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Started first pass

---

Reviewed 4 of 9 files at r1.
Review status: 4 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/examples/kuka_iiwa_arm/iiwa_common.h, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L-na7c0nkOuar71i9gq:-L-na7c0nkOuar71i9gr:b18cy4r) ([raw file](https://github.com/robotlocomotion/drake/blob/e6129a00a32fc91eb012de7ab3d6f83daa275fa0/drake/examples/kuka_iiwa_arm/iiwa_common.h#L22)):*
> ```objc
> // Kuka publishes these in degrees, they've been converted to radians here.
> // clang-format off
> const VectorX<double> kIiwaMaxJointVelocities =
> ```

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables requires that global variables be POD type.  (I think an array of doubles would probably be legal, as would a `get_iiwa_max_joint_velocities` function or defining this as a function local constant in `ApplyJointVelocityLimits` or something.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L-nb0pY8QIRzfiN286w:b-g6adnf)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 2 of 9 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[drake/examples/kuka_iiwa_arm/iiwa_common.h, line 22 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L-na7c0nkOuar71i9gq:-L055NEE__6E318R4yfV:bv3vu6e) ([raw file](https://github.com/robotlocomotion/drake/blob/e6129a00a32fc91eb012de7ab3d6f83daa275fa0/drake/examples/kuka_iiwa_arm/iiwa_common.h#L22)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables requires that global variables be POD type.  (I think an array of doubles would probably be legal, as would a `get_iiwa_max_joint_velocities` function or defining this as a function local constant in `ApplyJointVelocityLimits` or something.)
</blockquote></details>

I've gone with `get_iiwa_max_joint_velocities()` since I foresee this being useful in planning as well.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
@drake-jenkins-bot linux-xenial-clang-bazel-experimental-everything please

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L05hltbssoT413xZfAE:b-rs4zsh)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
First pass complete, added a few more comments.  Thanks for putting this together!  (and for figuring out how to break up the individual PRs)

---

Reviewed 10 of 18 files at r2.
Review status: 8 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05ZfbtJ3AAqKvqtH3u:-L05Zfbu0bYvmb_q93CT:b-b1kj4v) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L55)):*
> ```cpp
> };
> 
> // Generates a sequence (@p num_via_points + 1) of key frames s.t. the end
> ```

None of the parameters referred to with `@p` exist for this function.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 71 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05Zu5AUGQ_DZggHYuc:-L05Zu5Bqt30B7zGaJLd:b-d18zku) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L71)):*
> ```cpp
>   const double max_joint_position_change{request.max_joint_position_change};
> 
>   const int kNumKnots = times.size();
> ```

kConstant naming should only be used for variables whose value is fixed for the duration of the program.  http://drake.mit.edu/styleguide/cppguide.html#Constant_Names

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 73 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05_EhHbjVqqBan6O2A:-L05_EhHbjVqqBan6O2B:bwhdkhz) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L73)):*
> ```cpp
>   const int kNumKnots = times.size();
> 
>   int kNumJoints{robot->get_num_positions()};
> ```

see above re: kConstant naming

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 440 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05i05g7IOoQE6IqAfM:-L05i05hd7kp4ZellGmw:bz7tlk2) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L440)):*
> ```cpp
>   std::vector<double> yaw_offsets{M_PI, 0.0};
>   std::vector<double> pitch_offsets{M_PI / 8};
>   int kNumJoints = robot->get_num_positions();
> ```

see above re: kConstant syntax

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 452 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05i5Y5CaIChaJG1egU:-L05i5Y5CaIChaJG1egV:bz7tlk2) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L452)):*
> ```cpp
>       PickAndPlaceState::kApproachPlace,
>       PickAndPlaceState::kLiftFromPlace};
>   int kNumKnots = states.size() + 1;
> ```

see above re: kConstant syntax

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 634 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05jRzMaD8ULeV35cWg:-L05jRzNYzuds4oUQmry:bqn6vq5) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L634)):*
> ```cpp
>       } else {
>         double duration{kShortDuration};
>         double position_tolerance{tight_pos_tol_(0)};
> ```

Keeping these as vectors of 3 values seems a bit misleading if only one is used.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 758 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05lAwuPZ60To0kJBrW:-L05lAwuPZ60To0kJBrX:b9n0qpw) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L758)):*
> ```cpp
>           q.push_back(q_traj.value(t));
>         }
>         std::unique_ptr<RigidBodyTree<double>> robot{
> ```

BTW it seems a little excessive to build a new tree every time we create a new move action when `configuration_` never changes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L05mpb6Ks_Dr7VYwZvS:b-a91y75)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

SeanCurtis-TRI commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 71 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05Zu5AUGQ_DZggHYuc:-L05noKT4O3Vc1BLoJdL:bcvo19m) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L71)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

kConstant naming should only be used for variables whose value is fixed for the duration of the program.  http://drake.mit.edu/styleguide/cppguide.html#Constant_Names
</blockquote></details>

FYI For me own edification, as I read the guide, the `kVariableName` is *required* for values fixed over the duration and *optional* (but not precluded) for other const values (e.g., automatic storage duration). That would *allow* this usage, wouldn't it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 71 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05Zu5AUGQ_DZggHYuc:-L05sESsXpFXSPoY1my4:b-exe4ab) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L71)):*
<details><summary><i>Previously, SeanCurtis-TRI (Sean Curtis) wrote…</i></summary><blockquote>

FYI For me own edification, as I read the guide, the `kVariableName` is *required* for values fixed over the duration and *optional* (but not precluded) for other const values (e.g., automatic storage duration). That would *allow* this usage, wouldn't it?
</blockquote></details>

wow, I somehow completely failed to internalize that final sentence (I wonder if it's new or if I've been missing it for years?)  As it's written, I believe you're correct.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 71 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05Zu5AUGQ_DZggHYuc:-L05szs9mtB2eIpanXKQ:b6lwix0) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L71)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

wow, I somehow completely failed to internalize that final sentence (I wonder if it's new or if I've been missing it for years?)  As it's written, I believe you're correct.
</blockquote></details>

That's a misreading of the guide.

If a variable is static and a constant, it should use `kFoo`.

If a variable takes on more than one value for any given process (as this one appears to), then is _should not_ use `kFoo`.

If it has only one value for any given process, but is not marked `static`, then its author's discretion.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Comments addressed (Except for the one on the `PlanInterpolatingMotion()` comment, which I missed). PTAL.

---

Review status: 4 of 10 files reviewed at latest revision, 7 unresolved discussions.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 71 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05Zu5AUGQ_DZggHYuc:-L06DD8cpCDNmiroPC7w:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L71)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

That's a misreading of the guide.

If a variable is static and a constant, it should use `kFoo`.

If a variable takes on more than one value for any given process (as this one appears to), then is _should not_ use `kFoo`.

If it has only one value for any given process, but is not marked `static`, then its author's discretion.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 73 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05_EhHbjVqqBan6O2A:-L06DDZGWBUg3pqs0hqb:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L73)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

see above re: kConstant naming
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 440 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05i05g7IOoQE6IqAfM:-L06DE5IXbJJJI7kj2wy:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L440)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

see above re: kConstant syntax
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 452 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05i5Y5CaIChaJG1egU:-L06DEot3l2IwdWOko2z:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L452)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

see above re: kConstant syntax
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 634 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05jRzMaD8ULeV35cWg:-L06DlrRKrKv9kC_u3h3:b-33tjo2) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L634)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

Keeping these as vectors of 3 values seems a bit misleading if only one is used.
</blockquote></details>

They're used as vectors in `ComputeNominalConfigurations()`. Here we only need a a single value. I've changed this to `minCoeff()` so that we aren't arbitrarily selecting one of the values and assuming that all three values are equal.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 758 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05lAwuPZ60To0kJBrW:-L06F1tLsak6CHSl64U0:b-99vpvu) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L758)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

BTW it seems a little excessive to build a new tree every time we create a new move action when `configuration_` never changes.
</blockquote></details>

Agreed. I've modified `IiwaMove::MoveJoints` to take a vector of joint names rather than a `RigidBodyTree`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L06Qg2ElzezTmqxTFbc:b-h7z99)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
Reviewed 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

sammy-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> +@ggould-tri for platform review per rotation

---

Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L0AN_X73iQzUoHBl9_Y:bqwg0b7)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 55 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L05ZfbtJ3AAqKvqtH3u:-L0AR5o_TpbCQOwOLLaR:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/cb7bdf89f25023d3c956cbdeaed6779f0d8f52a1/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L55)):*
<details><summary><i>Previously, sammy-tri (Sam Creasey) wrote…</i></summary><blockquote>

None of the parameters referred to with `@p` exist for this function.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
Review status: 7 of 10 files reviewed at latest revision, 26 unresolved discussions.

---

*[examples/kuka_iiwa_arm/iiwa_common.cc, line 37 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0APwiEcdiYan7xNHTj:-L0Af0eYqFgQIS2HEjah:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/iiwa_common.cc#L37)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

The constants below should be commented as to where they came from (eg, page or section number of the iiwa documentation).

Additionally at author's discretion it might be clearer to state them with the computations that derived them (` // 85 deg/s in rad/s` for example) as not all users will recognize the value `1.483529` as an old friend.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/iiwa_common.h, line 57 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AOWlVQpRZkbN-z-QC:-L0Af22LNRbAE_pVFNt2:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/iiwa_common.h#L57)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

Now that the "maximum joint velocity specified" is not in the method signature, this comment is a bit surprising.  You should at least pluralize "velocity" and probably say "the robot's maximum joint velocity" rather than the less clear "specified".
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 82 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATLtGcMoe7K8keVh5:-L0ApAjNh4e2hIXEM9RU:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L82)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 94 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATPpqvQ64J0pZlWBF:-L0ApEmvK2U_Gct0owNa:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L94)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 111 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATTMLliXZetCKzvdm:-L0ApHuVs585Gw7Gu5Ua:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L111)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 124 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATUbRylsvqKq9fKkY:-L0ApLdppepKAx-X5s2M:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L124)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 163 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATbsS21WchdoW0RLa:-L0ApO0ekMIccIGdf7du:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L163)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 207 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0ATgi5gtCGl-wzALii:-L0ApQtOpPDSHKeh0DcJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L207)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 432 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUe6TdHYYq5MrPYBv:-L0ApTkwNliTbpuTV8xD:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L432)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 436 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUep2ZgRAeWfjrqUB:-L0ApUJh-n0izm21zSuz:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L436)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 444 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUfsnSIvFz_jHXoJ7:-L0ApUmwILkGUKGYc_Sj:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L444)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 449 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUgpMnCyyd5l2pDUy:-L0ApWlZTgRYS28R9ZnV:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L449)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 453 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUhgQcR9oKXxMNjHP:-L0Ap_9GpniG1nG6lIrz:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L453)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 458 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUiU4IOVLAgkRrynE:-L0Apa5ugNdxNjRxLmU5:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L458)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 476 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUkvMjPx9Kb7GoTDB:-L0ApfGKA2rdwsDtTBzf:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L476)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 500 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUpWBexCjVEpJkNmk:-L0Apgx_ebrVEgQIPuKD:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L500)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 516 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AUs5Z-hH4j-bLfzYu:-L0AuFoWKqofA6yZNDVz:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L516)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please reflow this comment.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 704 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AV6OFJ3Lyg9EJVltq:-L0ApqQUAUuMQmNczaAW:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L704)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 757 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AWOv59AJrA4pdnIxn:-L0AvFwMBmFcg88WbeOt:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L757)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW for all the cases below, our style guide prefers putting the `break;` inside the braces.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 786 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AV9pDpzQ8DAw_bvMW:-L0AprnTCmhdIysNqV-i:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L786)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 833 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVkBviIVHPW3nrHwF:-L0AvHe8DuSI1zT_t_ya:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L833)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW our style guide prefers putting the `break;` inside the braces.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 846 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVCAdWaUKElVcReb0:-L0AvqE0PqfWLb85PVAt:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L846)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW please punctuate this complete sentence.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 860 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVbcdqlEzBol6O5RG:-L0AvKp_La6fYwhj0ZjJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L860)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW our style guide prefers placing the `break;` inside the braces.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 886 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVq7fpSHd2aDfwY3n:-L0AvOxcZIatDzbK974-:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L886)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW our style guide prefers placing the `break;` inside the braces.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 895 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVvGqixLhqUd1U2Ua:-L0AvlazjELTHTvBOdQx:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L895)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW our style guide prefers placing the `break;` inside the braces.
</blockquote></details>

Done.

---

*[examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc, line 897 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-L0AVweajY614kFeXnDi:-L0Avlyo3pTyou6elWAE:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/bae7c5445f30b76371c7bab4cd64f3157db4e442/examples/kuka_iiwa_arm/pick_and_place/pick_and_place_state_machine.cc#L897)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

BTW our style guide prefers placing the `break;` inside the braces (but also does not require the braces at all in this case, which is allowed to be a one-liner if you prefer).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

ggould-tri commented:
Reviewed 3 of 3 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567)*
<!-- Sent from Reviewable.io -->

avalenzu commented:
-(status: curate commits before merging) 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7567#-:-L0BUb_voJ7QRUJ8dUT6:b-l70wzf)*
<!-- Sent from Reviewable.io -->
